dds/DCPS/TypeSupportImpl.h

```diff
@@ -36,6 +38,41 @@ template <typename Message> struct DDSTraits;
 
 template <typename Message> struct MarshalTraits;
 
+class OpenDDS_Dcps_Export SerializedSizeBound {
+public:
+  SerializedSizeBound()
+  : bounded_(false)
+  , bound_(0)
+  {
+  }
+
+  SerializedSizeBound(size_t bound)
+  : bounded_(true)
```

Comment:
If `bound` is 0, should this be `false`?

Reply:
If a 0 is passed in, then it's bounded to 0. Setting `bounded_` to `false` would make it unbounded.

Reply:
I guess I was thinking this was one of the cases where 0 indicates unbounded.  If it's not, nothing to do here...

Reply:
Yeah the reason behind this class to enforce the distinction, because the existence of that rule in addition to a separate bounded function was kinda weird.